tests/com.espressif.idf.ui.test/src/com/espressif/idf/ui/test/operations/ProjectTestOperations.java

```diff
@@ -413,4 +487,41 @@
 			// logging will be added to show no projects were found
 		}
 	}
+
+	public static void launchCommandUsingContextMenu(String projectName, SWTWorkbenchBot bot, String contextMenuLabel)
+	{
+		SWTBotTreeItem projectItem = fetchProjectFromProjectExplorer(projectName, bot);
+		if (projectItem != null)
+		{
+			projectItem.select();
+			projectItem.contextMenu(contextMenuLabel).click();
+		}
+	}
+
+	public static void waitForProjectClean(SWTWorkbenchBot bot) throws IOException
+	{
+		SWTBotView consoleView = viewConsole("Espressif IDF Tools Console", bot);
+		consoleView.show();
+		consoleView.setFocus();
+		TestWidgetWaitUtility.waitUntilViewContains(bot, "Done", consoleView, 0);
+	}
```

Comment:
@AndriiFilippov this is a valid point from coderabbitai

Reply:
@alirana01 since I am using 'joinJobByName' method which waits for a job with a specific name to complete, there is no need for an additional timeout.